tests/Integration/Persistence/ProxyPHP84TestCase.php

```diff
@@ -0,0 +1,192 @@
+        self::assertSame('default1', $object->getProp1());
+        self::assertFalse((new \ReflectionClass($object))->isUninitializedLazyObject($object));
+
+        self::ensureKernelShutdown();
+
+        $client = self::createClient();
+        $client->request('GET', "/{$this->dbms()}/update/{$object->id}");
+        self::assertResponseIsSuccessful();
+
+        self::assertTrue((new \ReflectionClass($object))->isUninitializedLazyObject($object));
+        self::assertSame('foo', $object->getProp1());
+        self::assertFalse((new \ReflectionClass($object))->isUninitializedLazyObject($object));
+    }
+
+    /**
+     * @test
+     * @requires PHP >= 8.4
+     * @depends it_can_refresh_objects_with_php84_proxies
+     */
+    #[Test]
+    #[RequiresPhp('>= 8.4')]
+    #[Depends('it_can_refresh_objects_with_php84_proxies')]
+    public function it_can_refresh_objects_with_php84_tracker_is_empty_after_test(): void
+    {
+        self::assertSame(0, PersistedObjectsTracker::countObjects());
+    }
+
+    /**
+     * @test
+     * @requires PHP >= 8.4
+     */
+    #[Test]
+    #[RequiresPhp('>= 8.4')]
+    public function deleting_an_object_does_not_create_a_refresh_error(): void
+    {
+        $object = $this->factory()->create();
+        self::assertSame('default1', $object->getProp1());
+        self::assertFalse((new \ReflectionClass($object))->isUninitializedLazyObject($object));
+
+        self::ensureKernelShutdown();
+
+        $client = self::createClient();
+        $client->request('GET', "/{$this->dbms()}/delete/{$object->id}");
+        self::assertResponseIsSuccessful();
+
+        self::assertTrue((new \ReflectionClass($object))->isUninitializedLazyObject($object));
+        self::assertNull($object->id);
+        self::assertFalse((new \ReflectionClass($object))->isUninitializedLazyObject($object));
+    }
+
+    /**
+     * @test
+     * @requires PHP >= 8.4
+     */
+    #[Test]
+    #[RequiresPhp('>= 8.4')]
+    public function it_can_refresh_the_same_object_multiple_times(): void
+    {
+        $object = $this->factory()->create();
+        self::ensureKernelShutdown();
+
+        $client = self::createClient();
+        $client->request('GET', "/{$this->dbms()}/update/{$object->id}");
+        self::assertResponseIsSuccessful();
+
+        $objectTracker = Configuration::instance()->persistedObjectsTracker;
+        self::assertNotNull($objectTracker);
+
+        $objectTracker->refresh();
+        $objectTracker->refresh();
+        $objectTracker->refresh();
+
+        self::assertTrue((new \ReflectionClass($object))->isUninitializedLazyObject($object));
+        self::assertSame('foo', $object->getProp1());
+        self::assertFalse((new \ReflectionClass($object))->isUninitializedLazyObject($object));
+    }
+
+    /**
+     * @test
+     * @requires PHP >= 8.4
+     */
+    #[Test]
+    #[RequiresPhp('>= 8.4')]
+    public function it_can_refresh_after_command_terminated(): void
```

Comment:
this test is only here to confirm that using `CommandTester` updates the objects, but since `ConsoleTerminate` nor `kernel.reset` are triggered, the proxy mechanism is not involved at all in this test :shrug: 

this could be problematic, because if the command would call `$em->clear()` or perform raw SQL, this test would fail :disappointed: 

I think there is nothing we can do, except documenting this, and/or modifying how `CommandTester` works, and make it trigger one of the event. WDYT?